docs/docs/training-data-format.mdx

```diff
@@ -1,355 +1,655 @@
+Once defined, the [happy path](../glossary#happy--unhappy-paths)
+for a form should be specified as a [rule](../core/forms), but interruptions of forms or 
+other "unhappy paths" should be included in stories so that the model can 
+generalize to unseen conversation sequences.
+As a step in a story, a form takes the following basic format:
+
 
+```yaml
+    steps:
+    - intent: find_restaurant
+    - action: restaurant_form <!--Activate the form-->
+    - form: restaurant_form   <!--This form is currently active-->
+    - form: null              <!--No form is active-->
+    - action: utter_restaurant_found
 ```
-Give me a [small]{"entity": "size", "group": "1"} pizza with [mushrooms]{"entity": "topping", "group": "1"} and
-a [large]{"entity": "size", "group": "2"} [pepperoni]{"entity": "topping", "group": "2"}
+
+
+The `action` step activates the form and begins looping over the required slots. The `form: restaurant_form`
+step indicates that there is a currently active form. Much like a `slot_was_set` step,
+a `form` step doesn't **set** a form to active but indicates that it should already be activated.
+In the same way, the  `form: null` step indicates that no form should be active before the subsequent
+steps are taken. 
+
+A form can be interrupted and remain active; in this case the interruption should come after the 
+`action: <form to activate>` step and be followed by the `form: <active form>` step. 
+An interruption of a form could look like this:
+
+```yaml
+  - rule: interrupted food
+    steps:
+      - intent: request_restaurant
+      - action: restaurant_form
+      - intent: chitchat
+      - action: utter_chitchat
+      - form: restaurant_form
+      - form: null
+      - action: utter_slots_values
 ```
 
-See [Training Data Format](./training-data-format) for details on how to define entities with roles and groups in your training data.
-
-The entity object returned by the extractor will include the detected role/group label.
-
-```json
-{
-  "text": "Book a flight from Berlin to SF",
-  "intent": "book_flight",
-  "entities": [
-    {
-      "start": 19,
-      "end": 25,
-      "value": "Berlin",
-      "entity": "city",
-      "role": "departure",
-      "extractor": "DIETClassifier",
-    },
-    {
-      "start": 29,
-      "end": 31,
-      "value": "San Francisco",
-      "entity": "city",
-      "role": "destination",
-      "extractor": "DIETClassifier",
-    }
-  ]
-}
+
+#### Slots
+
+**[Slots](../core/slots)** act as the bots memory. Slots are **set** by entities or by custom actions and **referenced** by stories in `slot_was_set` steps. For example.
+
+```yaml
+    steps:
+    - intent: celebrate_bot
+    - slot_was_set: feedback_value
+      value: positive
+    - action: utter_yay
 ```
 
+This means the story requires that the current value for the `feedback_value`
+slot be `positive` for the conversation to continue as specified.
```

Comment:
We don't have to add slot steps for unfeaturized slots, right?

Reply:
No, but I think that's explained on the slots types part of the domain.

Reply:
Format issue for slots though - I see two examples in notion:

```
    slot_was_set: 
    - context_scenario: holiday
    - holiday_name: thanksgiving
```
and
```
  - slot_was_set: feedback_value
    value: positive
```
In either case, what's the format for a text slot where value doesn't matter? Can it be just 
```
- slot_was_set: text_slot
```

Reply:
Mmm, no idea. @degiz ?

Reply:
The parser won't read that:
```
slot_was_set: 
- context_scenario: holiday
- holiday_name: thanksgiving
```

The only way is to use atm is:
```
- slot_was_set: feedback_value
  value: positive
```
and you can omit the `value`!

Reply:
Ok! I'll take it out of notion too then.

Reply:
I see where the confusion comes from :( 
Checkpoints have the slots, and it follows that syntax:
```
- checkpoint: greet_checkpoint
    # This checkpoint should only apply if slot is set to the specified value
    slot_was_set: 
    - context_scenario: holiday
    - holiday_name: thanksgiving
```

Reply:
oof. What if a checkpoint must be conditioned on a set slot with irrelevant value?